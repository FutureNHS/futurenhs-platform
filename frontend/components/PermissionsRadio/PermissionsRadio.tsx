import React, { HTMLProps } from "react";

import { Radios } from "nhsuk-react-components";
import FormGroup from "nhsuk-react-components/lib/util/FormGroup";
import { FormElementProps } from "nhsuk-react-components/lib/util/types/FormTypes";
import styled from "styled-components";

import { RoleRequired } from "../../lib/generated/graphql";

interface PermissionsRadioProps
  extends HTMLProps<HTMLInputElement>,
    FormElementProps {
  inputRef?: (inputRef: HTMLInputElement | null) => any;
}

const StyledRadios = styled(Radios)`
  .nhsuk-radios__label,
  .nhsuk-radios__hint {
    padding-left: 28px;
    padding-top: 0px;
  }
`;

const PermissionsRadio: React.FC<PermissionsRadioProps> = ({
  inputRef,
  hidden,
}) => (
  <div hidden={hidden}>
    <h4>Viewing permissions</h4>
    <FormGroup inputType="radios">
      {() => (
        <StyledRadios name="roleRequired" id="roleRequired">
          <Radios.Radio
            defaultChecked
            inputRef={inputRef}
            value={RoleRequired.PlatformMember}
            id="all-members"
            hint="This will make the folder and its contents visible to all members of FutureNHS"
          >
            Make visible to all FutureNHS members
          </Radios.Radio>
          <Radios.Radio
            inputRef={inputRef}
            value={RoleRequired.WorkspaceMember}
            id="workspace-members"
            hint="This will make the folder and its contents visible to members of your workspace."
          >
            Make visible to all members of this workspace
          </Radios.Radio>
        </StyledRadios>
      )}
    </FormGroup>
  </div>
);

export default PermissionsRadio;
