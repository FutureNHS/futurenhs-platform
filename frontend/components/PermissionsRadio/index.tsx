import Permissions from "./PermissionsRadio";

export { Permissions };
