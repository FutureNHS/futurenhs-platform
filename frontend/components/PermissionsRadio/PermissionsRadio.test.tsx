import React from "react";

import { render } from "@testing-library/react";

import PermissionsRadio from "./PermissionsRadio";

describe(PermissionsRadio, () => {
  it("renders matching snapshot", () => {
    const container = render(<PermissionsRadio />);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
