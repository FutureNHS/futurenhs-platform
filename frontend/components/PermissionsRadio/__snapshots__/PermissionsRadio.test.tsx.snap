// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermissionsRadio renders matching snapshot 1`] = `
<DocumentFragment>
  .c0 .nhsuk-radios__label,
.c0 .nhsuk-radios__hint {
  padding-left: 28px;
  padding-top: 0px;
}

<div
    class="nhsuk-form-group"
  >
    <div
      class="nhsuk-form-group"
    >
      <div
        class="nhsuk-radios c0"
        id="permissions"
        role="radiogroup"
      >
        <div
          class="nhsuk-radios__item"
        >
          <input
            aria-describedby="permissions-1--hint"
            aria-labelledby="permissions-1--label"
            class="nhsuk-radios__input"
            id="permissions-1"
            name="permissions"
            type="radio"
            value="all-members"
          />
          <label
            class="nhsuk-label nhsuk-radios__label"
            for="permissions-1"
            id="permissions-1--label"
          >
            Make visible to all FutureNHS members
          </label>
          <span
            class="nhsuk-hint nhsuk-radios__hint"
            id="permissions-1--hint"
          >
            This will make the folder and its contents visible to all members of FutureNHS
          </span>
        </div>
        <div
          class="nhsuk-radios__item"
        >
          <input
            aria-describedby="permissions-2--hint"
            aria-labelledby="permissions-2--label"
            class="nhsuk-radios__input"
            id="permissions-2"
            name="permissions"
            type="radio"
            value="workspace-members"
          />
          <label
            class="nhsuk-label nhsuk-radios__label"
            for="permissions-2"
            id="permissions-2--label"
          >
            Make visible to all members of this workspace
          </label>
          <span
            class="nhsuk-hint nhsuk-radios__hint"
            id="permissions-2--hint"
          >
            This will make the folder and its contents visible to members of your workspace.
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
